crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -593,15 +642,58 @@ impl SettlementEncoder {
             },
         );
 
+        // We trampoline the custom user interactions over the
+        // `MultiSendCallOnly` contract, ensuring that we aren't calling user-
+        // specified calls from a privileged context (i.e. the Settlement
+        // contract). We also collapse all `CoWSwapEthFlow::wrapAll` calls into
+        // one for gas efficiency.
+        let group_interactions = |interactions: Vec<InteractionData>| {
```

Comment:
I think I'd rename this to `custom_interactions`. Not entirely sure what the idea behind `group_interactions` was.

Reply:
Left over of a refactor, never renamed the closure.

Reply:
2e254aff6ea549cf7ff4cd65cb60a581dc0ab76f